File: 1492_The_Kth_Factor_Of_N.cpp
class Solution {
public:
    int kthFactor(int n, int k) {
        vector<int> factors = {1}; 
        int i = 2;
        while( i <= n){
            if( n % i == 0){
                factors.push_back(i); 
            }
            i++;
        }
        if(k <= factors.size())
        {
            return factors[k-1]; 
        }
        return -1; 
    }
};

File: 1492_The_Kth_Factor_Of_N.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        factors = [1]
        i = 2
        while i <= n :
            if n % i == 0:
                factors.append(i)
            i += 1 
        if k <= len(factors):
            return factors[k - 1]
        return -1
        